### DISCUSSION TITLE
REGION is different when viewed from different websites

### DISCUSSION THREAD

**Discussion by fritzisa** [code4recovery/12-step-meeting-list#1078] [Q&A] [answered]
**Dear friends,**

The REGION isn't the same when viewed from two different sites.

When filtering for the meeting Morton, the REGION is different on SEPIA website (1) than Area 59 website (2)

(1) https://aasepia.org/?search=morton

(2) https://area59aa.org/find-a-meeting/?search=morton

Area 59 uses the feed provided by SEPIA.

?? Is there a reason for this inconsistency ??

Thank you in advance.

**Scott Fritzinger**  |  Area 59 Web Servant  |  (610) 762-7792

![Morton-SEPIA](https://user-images.githubusercontent.com/95769327/219121834-30c5bf58-7586-4e0a-8369-33e2d6f78036.jpg)

![Morton-A59](https://user-images.githubusercontent.com/95769327/219121864-74fea18c-c64a-47d7-8201-deb6ec51747d.jpg)

**Comment by joshreisner** [comment]
hi @fritzisa my strong suspicion is that TSML doesn't actually support importing the `regions` key that SEPIA is using, and that it falls back to the geocoded city. 

most TSML sites use `region` and `sub_region` which i think is what TSML is currently supporting.

we'll file a bug.

sepia data:
```
  {
    "name": "Morton",
    "slug": "meeting-2545-franklin-ave-clifton-heights-pa-19018-usa-6-100000",
    "notes": "",
    "updated": "2020-08-03 16:00:48",
    "url": "https://www.aasepia.org/meetings/?meeting=meeting-2545-franklin-ave-clifton-heights-pa-19018-usa-6-100000",
    "day": 6,
    "time": "10:00",
    "end_time": "11:00",
    "conference_url": "",
    "conference_phone": "",
    "types": [
      "O",
      "BE"
    ],
    "location": "Holy Spirit Lutheran Church (D32)",
    "formatted_address": "2545 Franklin Ave, Clifton Heights, PA 19018, USA",
    "latitude": "39.9133542",
    "longitude": "-75.3106973",
    "regions": [
      "Delaware County",
      "Clifton Heights"
    ],
    "group": "D32 / GSO #140549",
    "feedback_url": "https://aasepia.org/meeting-status/"
  },
```

area 59 JSON:
```
  {
    "id": 2071702,
    "name": "Morton",
    "slug": "meeting-2545-franklin-ave-clifton-heights-pa-19018-usa-6-100000",
    "updated": "2020-08-03 20:00:48",
    "location_id": 2070691,
    "url": "https://area59aa.org/meetings/meeting-2545-franklin-ave-clifton-heights-pa-19018-usa-6-100000/",
    "day": 6,
    "time": "10:00",
    "end_time": "11:00",
    "time_formatted": "10:00 AM",
    "edit_url": "https://area59aa.org/wp-admin/post.php?post=2071702&action=edit",
    "types": [
      "O",
      "BE"
    ],
    "location": "Holy Spirit Lutheran Church (D32)",
    "location_url": "https://area59aa.org/locations/holy-spirit-lutheran-church-d32/",
    "formatted_address": "2545 Franklin Ave, Secane, PA 19018, USA",
    "approximate": "no",
    "latitude": 39.9133542,
    "longitude": -75.3106973,
    "region_id": 691,
    "region": "Sellersville",
    "regions": [
      "Sellersville"
    ],
    "group_id": 2071701,
    "group": "D32 / GSO #140549",
    "attendance_option": "in_person"
  },
```

**Reply by joshreisner** [reply]
what i don't understand by that logic is why the region is Sellersville; looks like it should be Secane

also when i [imported it to our demo site](https://code4recovery.org/meetings/meeting-2545-franklin-ave-clifton-heights-pa-19018-usa-6-100000?tsml-day=any&tsml-query=morton) the region is Philadelphia?

**Reply by tim-rohrer** [reply]
I was working on a reply when the neighbor came by.

I was wondering if Regions were defined in the Area 59 site?

**Comment by fritzisa** [comment]
Josh and Tim,

I don’t believe Area 59 defined the Regions.

Scott Fritzinger  |  Area 59 Web Servant  |  (610) 762-7792

**Comment by kiyote33** [comment, marked as answer]
Hey @fritzisa, we'll have a fix for this issue in the upcoming 3.15 major release for TSML. In the meantime, you could drop the following code in your (child) theme functions.php file to resolve this problem until 3.15 is ready: 
```
if (!function_exists('tsml_import_reformat')) {
	function tsml_import_reformat($imported_data) {
		$meetings = array();

		foreach ($imported_data as $meeting) { 
			
			if (is_array($meeting['regions']) !== false) {
				$meeting['region'] = $meeting['regions'][0];
				if (count($meeting['regions']) > 1) {$meeting['sub_region'] = $meeting['regions'][1];}	    
			}			
                        array_push($meetings, $meeting);
		}
		return $meetings;
	} 
}
```
